### ISSUE TITLE
Allow sequencing evaluations relative to io actions

### ISSUE THREAD

**Issue by Borgvall** [haskell/parallel#1]
It is sometimes useful to sequence the evaluation relative to IO actions. For instance one might spark a computation in parallel, while the main thread is waiting for further input.

I have written a (ghc only) patch that provides functions that allow this (https://github.com/Borgvall/parallel/commit/3d5b0fbc29cc0e388d1b04f70d4181771960cc3f). You might consider merging it.

The downside is, it depends on the internal representation of IO and Eval, and adds functions to the external interface.

**Comment by Borgvall** [comment]
I have added a test for runEvalIO (https://github.com/Borgvall/parallel/commit/7ee10268f6bf6e4ad71309f659428d18be9cbbd8) and implement it using the classic Eval implementation using `Done` (https://github.com/Borgvall/parallel/commit/563bb87a2b4a42ff9e8e7c5e857aaf399c7a6fa1).

**Comment by treeowl** [comment]
I think there's a much simpler way to accomplish this in vaguely recent compiler versions:

```haskell
{-# Language GeneralizedNewtypeDeriving #-}
newtype Eval a = Eval (IO a) deriving (Functor, Applicative, Monad)
runEval (Eval m) = unsafePerformIO m
```
Exposing the newtype constructor from an "internals" module would give users maximal flexibility, but barring that, exposing wrapping and unwrapping functions seems reasonable.

**Comment by Borgvall** [comment]
If you propose to change the internal implementation of `Eval`, please open a new issue. It would have a much greater impact, and needs broader discussions.

I think, that sequencing evaluations relative to io actions should be part of the external interface.

The reverse (`IO -> Eval`) would break the basic idea of the parallel package!

**Comment by Borgvall** [comment]
Fixed by the merge of #3.
